modules/lsp.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  config,
+  ...
+}:
+let
+  inherit (lib) mkOption types;
+
+  cfg = config.lsp;
+
+  serverType = types.submodule {
+    options = {
+      enable = lib.mkEnableOption "the language server";
+
+      config = mkOption {
+        type = with types; attrsOf anything;
```

Comment:
Perhaps going forward we should consider using `(pkgs.formats.lua { }).type`?

This is a bit more restrictive than `anything`: https://github.com/NixOS/nixpkgs/blob/55024d64cdf0cd2a0a2764857962a1da28a6ee23/pkgs/pkgs-lib/formats.nix#L722-L740

```nix
nullOr (oneOf [
  bool
  float
  int
  path
  str
  luaInline
  (attrsOf valueType)
  (listOf valueType)
])
```

That said, it only supports our non-standard "raw lua" by accident; `attrsOf str` _happens_ to match `{ __raw = ""; }`...

We should probably have this discussion outside of this PR, as using `attrsOf anything` is consistent with our current approach throughout nixvim.